src/node_crypto.h

```diff
@@ -339,6 +339,161 @@ class SSLWrap {
   friend class SecureContext;
 };
 
+// A helper class representing a read-only byte array. When deallocated, its
+// contents are zeroed.
+class ByteSource {
+ public:
+  ByteSource() = default;
+  ByteSource(ByteSource&& other);
+  ~ByteSource();
+
+  ByteSource& operator=(ByteSource&& other);
+
```

Comment:
If you want this class to only have move semantics, you should delete the copy constructor and the copy assignment operators
```suggestion
  ByteSource(ByteSource&) = delete;
  ByteSource& operator=(ByteSource&) = delete;
```

Reply:
If I understand your design, and your intent is for this class to only have by-value semantics, you can also delete the `new` operator.